File: src/dependencies/crazyflie_hardware/src/crtp_cpp/src/crtp_cpp/packer/console_packer.cpp
#include "crtp_cpp/packer/console_packer.hpp"

ConsolePacker::ConsolePacker()
    : CrtpPacker(PORT_CONSOLE) {}

CrtpPacket ConsolePacker::prepare_packet(const std::vector<uint8_t> &data)
{
    return CrtpPacker::prepare_packet(CHANNEL_CONSOLE, data);
}

CrtpPacket ConsolePacker::consolepacket()
{
    std::vector<uint8_t> data; // Empty vector for struct.pack("<")
    return prepare_packet(data);
}

File: src/dependencies/crazyflie_hardware/src/crtp/crtp/packers/console_packer.py
from .packer import Packer
from .crtp_packer import CrtpPacker

import struct

from typing import Callable


class ConsolePacker(Packer):
    PORT_CONSOLE = 0

    CHANNEL_CONSOLE = 0

    def __init__(self, crtp_packer_factory: Callable[[int], CrtpPacker]):
        super().__init__(crtp_packer_factory, self.PORT_CONSOLE)

    def consolepacket(self):
        data = struct.pack("<")
        return self._prepare_packet(channel=self.CHANNEL_CONSOLE, data=data)
